lib/response.js

```diff
@@ -447,8 +447,11 @@ internals.Response.prototype._marshal = function (next) {
 internals.Response.prototype._streamify = function (source, next) {
 
     if (source instanceof Stream) {
-        var stream = (source.socket || source);
-        if (stream._readableState.objectMode) {
+        if (typeof source._read != 'function' || typeof source._readableState != 'object') {
```

Comment:
Unfortunately not. We need to rely on duck typing, to support the common case of `Readable` streams created through the `readable-stream` module which are not an instance of `require('stream').Readable`.